accord-core/src/main/java/accord/local/cfk/SafeCommandsForKey.java

```diff
@@ -55,12 +55,13 @@ void updatePruned(SafeCommandStore safeStore, Command nextCommand, NotifySink no
     public void update(SafeCommandStore safeStore, Command nextCommand)
     {
         CommandsForKey prevCfk = current();
-        update(safeStore, nextCommand, prevCfk, prevCfk.update(nextCommand));
+        CommandsForKeyUpdate cfkUpdate = prevCfk.update(nextCommand);
+        update(safeStore, nextCommand, prevCfk, cfkUpdate);
     }
 
     private void update(SafeCommandStore safeStore, @Nullable Command command, CommandsForKey prevCfk, CommandsForKeyUpdate updateCfk)
     {
-        update(safeStore, command, prevCfk, updateCfk, NotifySink.DefaultNotifySink.INSTANCE);
+        update(safeStore, command, prevCfk, updateCfk, safeStore.replay() ? NotifySink.NoOpSink.INSTANCE : NotifySink.DefaultNotifySink.INSTANCE);
```

Comment:
I think we can switch behaviour more cleanly here, but also we need to either capture the notifications OR we need to actually submit them OR we need to avoid updating unmanaged altogether and batch notify them at the end - otherwise they may never be notified.

Reply:
Made it that we do submit notifications for now. The problem was that notifications were causing submission of new tasks.